android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java

```diff
@@ -3,44 +3,192 @@
 import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
-import android.content.IntentFilter;
-import android.content.BroadcastReceiver;
-import android.os.BatteryManager;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.Settings;
 import android.view.WindowManager;
 
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
-import com.facebook.react.bridge.WritableMap;
-import com.facebook.react.bridge.WritableNativeMap;
 import com.facebook.react.bridge.ReactMethod;
-import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
-import com.facebook.react.bridge.LifecycleEventListener;
 
-import javax.annotation.Nullable;
 
 public class ScreenBrightnessModule extends ReactContextBaseJavaModule {
+    public static final String MODULE_NAME = "ScreenBrightness";
+
+    private static final String PERMISSION_EVENT_NAME = "screenBrightnessPermission";
+    private static final int BRIGHTNESS_MAX = 255;
+    private static final int BRIGHTNESS_MIN = 0;
+
+    private final int mWriteSettingsRequestCode;
+
     private Activity mActivity;
 
-    public ScreenBrightnessModule(ReactApplicationContext reactApplicationContext, Activity activity) {
+    public ScreenBrightnessModule(
+            ReactApplicationContext reactApplicationContext,
+            Activity activity,
+            final int writeSettingsRequestCode) {
         super(reactApplicationContext);
         mActivity = activity;
+        mWriteSettingsRequestCode = writeSettingsRequestCode;
     }
 
     @Override
     public String getName() {
-        return "ScreenBrightness";
+        return MODULE_NAME;
+    }
+
+    /**
+     * Called by the main activity when the ACTION_MANAGE_WRITE_SETTINGS result is received.
+     * Emits the result into the JS context.
+     */
+    public void onPermissionResult() {
+        WritableMap payload = new WritableNativeMap();
+        boolean hasPermission = Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                Settings.System.canWrite(getReactApplicationContext());
+
+        payload.putBoolean("hasPermission", hasPermission);
+
+        getReactApplicationContext()
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                .emit(PERMISSION_EVENT_NAME, payload);
+    }
+
+    /**
+     * Returns whether the device has granted the application permission to write settings.
+     *
+     * @return True if WRITE_SETTINGS are granted.
+     */
+    public boolean hasSettingsPermission() {
```

Comment:
Does this need public visibility? I guess you could argue that it's something we want on the public api.